kmcd-props.md

```diff
@@ -326,6 +354,50 @@ The property checks if a successful `Pot . join` is preceded by a `TimeStep` mor
     rule derive(zeroTimePotInterestEnd, LogNote( _ , Pot . drip   )) => zeroTimePotInterest
 ```
 
+### Flap dai consistency
+
+```k
+    syntax ViolationFSM ::= "flapDaiConsistency"
+ // --------------------------------------------
+    rule derive(flapDaiConsistency, FlapKick(_, _, _, _)) => Violated(flapDaiConsistency) requires notBool(flapDaiGtOrEtSumOfFlapLots())
+    rule derive(flapDaiConsistency, FlapTend(_, _, _))    => Violated(flapDaiConsistency) requires notBool(flapDaiGtOrEtSumOfFlapLots())
+    rule derive(flapDaiConsistency, FlapDeal(_))          => Violated(flapDaiConsistency) requires notBool(flapDaiGtOrEtSumOfFlapLots())
+    rule derive(flapDaiConsistency, FlapYank(_))          => Violated(flapDaiConsistency) requires notBool(flapDaiGtOrEtSumOfFlapLots())
+```
+
+### Flap MKR consistency
+
+```k
+    syntax ViolationFSM ::= "flapMkrConsistency"
+ // --------------------------------------------
+    rule derive(flapMkrConsistency, FlapKick(_, _, _, _)) => Violated(flapMkrConsistency) requires notBool(flapMkrGtOrEtSumOfFlapBids())
+    rule derive(flapMkrConsistency, FlapTend(_, _, _))    => Violated(flapMkrConsistency) requires notBool(flapMkrGtOrEtSumOfFlapBids())
+    rule derive(flapMkrConsistency, FlapDeal(_))          => Violated(flapMkrConsistency) requires notBool(flapMkrGtOrEtSumOfFlapBids())
+    rule derive(flapMkrConsistency, FlapYank(_))          => Violated(flapMkrConsistency) requires notBool(flapMkrGtOrEtSumOfFlapBids())
+```
+
+### Flap Invariants
+
+```k
+    syntax Bool ::= flapDaiGtOrEtSumOfFlapLots() [function, functional]
+ // -------------------------------------------------------------------
+    rule [[ flapDaiGtOrEtSumOfFlapLots() =>
+              #lookup(USERDAI, Flap) >=Rat sumOfAllFlapLots(FLAP_BIDS)
+         ]]
```

Comment:
I would recommend:

```
    rule [[ flapDaiGtOrEtSumOfFlapLots() => #lookup(USERDAI, Flap) >=Rat sumOfAllFlapLots(FLAP_BIDS) ]]
```

or

```
    rule [[ flapDaiGtOrEtSumOfFlapLots()
         => lookup(USERDAI, Flap) >=Rat sumOfAllFlapLots(FLAP_BIDS)
         ]]
```

For consistency with this and other semantics.

Reply:
Done. Btw what does the `[[ ]]` actually mean here? I cargo-culted it from other code but I'm not sure what it's doing...